### DISCUSSION TITLE
Unable to manually change codec or bluetooth profile

### DISCUSSION THREAD

**Discussion by rileyav3** [Arkq/bluez-alsa#552] [Q&A]
Sorry if this is something really obvious but I am a total noob to linux/command line stuff. I a using moodeaudio to connect to my aptx supported bluetooth speaker via bluez. I followed a tutorial online to install aptx codecs as well as aac & LDAC and bluez now shows them as supported codecs, and even uses aac when I connect my phone as a source. 

But when I connect my speaker, SBC is still the used codec. No problem I will just change it manually with 'sudo bluealsa -c aptX'  , but upon entering this line I get returned 'bluealsa: E: Couldn't acquire D-Bus name. Please check D-Bus configuration. Requested name: org.bluealsa'

This also happens if I try to issue a command to select a bluetooth profile. Does anyone know what I am doing wrong?

**Comment by paul-1** [comment]
bluealsa is the daemon, bluealsa-cli (or dbus) is used to control bluealsa.   

SBC, AAC are enabled by default, so if you want to use aptX, then it is correct to add -c aptX to the blueasla command line.   But that just makes codecs available for use. (BTW.. SBC is a mandatory codec, so you cannot disable that one.)

So start bluealsa (With -c aptX plus any additional that you want), then connect your device.  Now use blueasla-cli to look at the device

use ```bluealsa-cli list-pcms``` to find the pcm device for your speaker.
use ```bluealsa-cli info <pcm-path>``` to find information about your speaker.  Note:  You should see here a list of supported codecs by your device.
then use ```bluealsa-cli codec <pcm-path> aptX``` to attempt to switch your device to aptX.

**Reply by rileyav3** [reply]
huh.. '-bash: bluealsa-cli: command not found' is all i get. now i'm even more confused!

**Comment by paul-1** [comment]
Are you building from source?   It has to be enabled when you build bluealsa with `—enable-cli`

see:  https://github.com/Arkq/bluez-alsa/wiki/Installation-from-source

**Reply by rileyav3** [reply]
oh, no, as mentioned I'm using moodeaudio which comes with bluez preinstalled and ready to use. so presumably the build i'm using doesn't have the cli function enabled? would the only thing to do be reinstalling bluez from source and ensuring those options are enabled?

sorry this is pretty over my head lol

**Reply by borine** [reply]
I think it would be best for you to raise this as an issue with moode audio, as that is where you obtained bluealsa from. Which codecs are available, and how to enable them, is entirely dependent on how bluealsa was built, and which version of the sources it was built from. If you wish to build it yourself from the original sources here then I am sure you will find help here with bluealsa-specific issues. However, if you have no previous experience of building software yourself on linux then it may be best to look for help from moode developers and users first. I'm not a moode user, so don't know which codecs or tools they have included - it may be that aptX is not included. The "online tutorial" you followed, is that specifically for moode?

**Reply by rileyav3** [reply]
understood, thanks for the help anyway. yes it was specifically for moode, [the link is here](https://steveblock.com/moode-bluetooth-codecs/)

the guide doesn't appear to require you to enable the codecs manually but as i said, mine is SBC by default
